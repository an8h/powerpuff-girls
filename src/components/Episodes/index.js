/* eslint-disable react/no-array-index-key */
// @flow

/* 
Episodes Component: 
It is shown when the data of each episode are fetched at Home page.
Rendered UI:
Table       : A table showing a list of episodes
name        : The title of the episode
airdate     : The date which the episode was on air
*/

import React, { useState } from "react";
import { Redirect } from "react-router-dom";
import { Table } from "reactstrap";

const Episodes = (props: Object) => {
  const { episodes } = props;
  const [clicked, handleClick] = useState(false);
  const [clickedItem, setClickedItem] = useState();

  return (
    <div className="table-container">
      <Table>
        <thead>
          <tr>
            <th>Number</th>
            <th>Date</th>
            <th>Title</th>
          </tr>
        </thead>
        <tbody>
          {episodes.map((episode, index) => (
            <tr
              key={index}
              onClick={() => {
                handleClick(true);
                setClickedItem(episode);
              }}
            >
              <th scope="row">{index + 1}</th>
              <td>{episode.airdate}</td>
              <td>{episode.name}</td>
            </tr>
          ))}
        </tbody>
      </Table>
      {clicked ? (
        <Redirect
          push
          to={{
            pathname: "/detail",
            state: { clickedItem },
          }}
        />
      ) : null}
    </div>
  );
};

export default Episodes;
