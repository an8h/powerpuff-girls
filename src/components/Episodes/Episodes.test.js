import React from "react";
import renderer from "react-test-renderer";
import Episodes from "./index";

const props = {
  episodes: [
    { name: "This is a test name", airdate: "2020-01-03" },
    { name: "This is a test name", airdate: "2020-01-03" },
  ],
};

describe("Episodes Component", () => {
  let component;

  beforeEach(() => {
    component = renderer.create(<Episodes episodes={props.episodes} />);
  });

  it("should render correctly", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
