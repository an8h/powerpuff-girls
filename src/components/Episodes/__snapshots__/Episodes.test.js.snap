// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episodes Component should render correctly 1`] = `
<div
  className="table-container"
>
  <table
    className="table table-striped"
  >
    <thead>
      <tr>
        <th>
          Number
        </th>
        <th>
          Date
        </th>
        <th>
          Title
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        onClick={[Function]}
      >
        <th
          scope="row"
        >
          1
        </th>
        <td>
          2020-01-03
        </td>
        <td>
          This is a test name
        </td>
      </tr>
      <tr
        onClick={[Function]}
      >
        <th
          scope="row"
        >
          2
        </th>
        <td>
          2020-01-03
        </td>
        <td>
          This is a test name
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
